src/permissions.psm1

```diff
@@ -0,0 +1,448 @@
+# Helper
+function Resolve-SafeguardAccessPolicyId
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$AccessPolicy
+    )
+
+    $ErrorActionPreference = "Stop"
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    if (-not ($AccessPolicy -as [int]))
+    {
+        try
+        {
+            $local:AccessPolicies = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET AccessPolicies `
+                                 -Parameters @{ filter = "Name ieq '$AccessPolicy'" })
+        }
+        catch
+        {
+            Write-Verbose $_
+            Write-Verbose "Caught exception with ieq filter, trying with q parameter"
+            $local:AccessPolicies = (Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET AccessPolicies `
+                                 -Parameters @{ q = $AccessPolicy })
+        }
+        if (-not $local:AccessPolicies)
+        {
+            throw "Unable to find access policy matching '$AccessPolicy'"
+        }
+        if ($local:AccessPolicies.Count -ne 1)
+        {
+            throw "Found $($local:AccessPolicies.Count) access policies matching '$AccessPolicy'"
+        }
+        $local:AccessPolicies[0].Id
+    }
+    else
+    {
+        $AccessPolicy
+    }
+}
+
+function Resolve-SafeguardRoleId
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Role
```

Comment:
Change Role to Entitlement.  We want to use Entitlement everywhere instead of Role.  Eventually this will change in the API.